src/basic/src/ScalarFunction.C

```diff
@@ -35,8 +38,27 @@ BaseScalarFunction<V, M>::BaseScalarFunction(const char * prefix,
     const VectorSet<V, M> & domainSet)
   : m_env(domainSet.env()),
     m_prefix((std::string)(prefix) + "func_"),
-    m_domainSet(domainSet)
+    m_domainSet(domainSet),
+#ifndef DISABLE_BOOST_PROGRAM_OPTIONS
+    m_parser(new BoostInputOptionsParser(m_env.optionsInputFileName())),
+#endif
+    m_fdStepSize(QUESO_BASESCALARFN_FD_STEPSIZE_ODV)
 {
+  // Snarf fd step size from input file.
+#ifndef DISABLE_BOOST_PROGRAM_OPTIONS
+  m_parser->registerOption<double>(m_prefix + "fdStepSize",
+                                   QUESO_BASESCALARFN_FD_STEPSIZE_ODV,
+                                   "step size for finite difference");
+  m_parser->scanInputFile();
+  m_parser->getOption<double>(m_prefix + "fdStepSize", m_fdStepSize);
+#else
+  m_fdStepSize = m_env.input()(m_prefix + "fdStepSize",
+                               QUESO_BASESCALARFN_FD_STEPSIZE_ODV);
+#endif
```

Comment:
I could make it a vector instead.  That way if the user sets it with a scalar value, it just sets every element.  Or are you thinking something even more flexible where the user can set every element with a scalar value and then override ones for certain components individually?